metadata_store_scripts/V2025.07.22.0__add_generate_masking_parameters_procedure.sql

```diff
@@ -0,0 +1,391 @@
+/*
+ * generate_masking_parameters - Generates masking parameters for Azure Data Factory dataflows.
+ *
+ * This stored procedure replaces the ADF dataflows generating masking parameters
+ * for improved performance and maintainability. It generates all necessary parameters for masking
+ * a specific table using Delphix Compliance Services APIs within Azure Data Factory pipelines.
+ *
+ * PARAMETERS:
+ * @source_database - Source database name
+ * @source_schema - Source schema name
+ * @source_table - Source table name
+ * @column_width_estimate - Estimated column width for batch calculations (default: 1000)
+ * @filter_key - Optional filter key for conditional masking (default: '')
+ * @dataset - Dataset type identifier (default: 'AZURESQL')
+ *
+ * OPERATING MODES:
+ * 1. Standard Mode (@filter_key = ''):
+ *   - Generates standard masking parameters for all columns with assigned algorithms
+ *   - Uses unconditional algorithms and default date formats
+ *
+ * 2. Conditional Mode (@filter_key provided):
+ *   - Applies conditional masking based on key column values
+ *   - Uses alias-specific algorithms and date formats
+ *   - Filters rules to match the specified condition alias
+ *
+ * OUTPUT PARAMETERS:
+ * - FieldAlgorithmAssignments: JSON mapping of encoded column names to masking algorithms
+ * - ColumnsToMask: JSON array of column names requiring masking
+ * - DataFactoryTypeMapping: ADF schema definition for API response parsing
+ * - NumberOfBatches: Calculated batch count for optimal processing
+ * - TrimLengths: JSON array of column max lengths for output trimming
+ * - DateFormatAssignments: JSON mapping of columns to their date format strings
+ * - ColumnsToCastAsStrings: JSON array of columns requiring string casting
+ * - ColumnsToCastBackTo*: JSON arrays of columns to cast back to specific ADF types
+ *
+ * DEPENDENCIES:
+ * - discovered_ruleset table: Contains masking rules and algorithm assignments
+ * - adf_type_mapping table: Maps database types to ADF types
+ * - Supports conditional masking via JSON structures in assigned_algorithm column
+ *
+ * FALLBACK BEHAVIOR:
+ * Always returns exactly one record. If no masking rules exist, returns default/empty
+ * values to prevent ADF lookup activity failures.
+ *
+ * NOTES:
+ * - Empty JSON arrays are converted to [""] for ADF UI compatibility
```

Comment:
question: I don't understand this, isn't `[]` as the passed value permissible and accurate?